### PR TITLE
Support query string for xref.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @ansyral @hellosnow @superyyrrzz @qinezh

### CODE REVIEW

File: Documentation/tutorial/links_and_cross_references.md [hunk 1]
```diff
@@ -267,3 +267,23 @@ When creating documentation for your own .NET library, it is needed to link to t
...
 
 > You can also manually create `xrefmap.yml` for project that is not generated by DocFX to be able to cross reference topics in it.
+
+## Advance: more options for cross reference
+
+You can create a cross link with following options:
+
+- `text`: the display text when the cross reference is has resolved correctly.
+
+  e.g.: `@"System.String?text=string"` will be resolve as @"System.String?text=string".
```
Comment (qinezh) [inline Documentation/tutorial/links_and_cross_references.md]: >resolve [](start = 47, length = 7)

resolved?

File: Documentation/tutorial/links_and_cross_references.md [hunk 2]
```diff
@@ -267,3 +267,23 @@ When creating documentation for your own .NET library, it is needed to link to t
 This can be downloaded on [nuget](https://www.nuget.org/packages/msdn.4.5.2).
 
 > You can also manually create `xrefmap.yml` for project that is not generated by DocFX to be able to cross reference topics in it.
+
+## Advance: more options for cross reference
```
Comment (superyyrrzz) [inline Documentation/tutorial/links_and_cross_references.md]: >Advance [](start = 3, length = 7)

Advanced

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by qinezh** [comment]
:shipit:

File: Documentation/tutorial/links_and_cross_references.md [hunk 3]
```diff
@@ -267,3 +267,23 @@ When creating documentation for your own .NET library, it is needed to link to t
...
+
+You can create a cross link with following options:
+
+- `text`: the display text when the cross reference is has resolved correctly.
+
+  e.g.: `@"System.String?text=string"` will be resolve as @"System.String?text=string".
+- `alt`: the display text when the cross reference do not have a `href` property.
```
Comment (hellosnow) [inline Documentation/tutorial/links_and_cross_references.md]: >do [](start = 51, length = 2)

does?

File: Documentation/tutorial/links_and_cross_references.md [hunk 4]
```diff
@@ -267,3 +267,23 @@ When creating documentation for your own .NET library, it is needed to link to t
 This can be downloaded on [nuget](https://www.nuget.org/packages/msdn.4.5.2).
 
 > You can also manually create `xrefmap.yml` for project that is not generated by DocFX to be able to cross reference topics in it.
+
+## Advance: more options for cross reference
+
+You can create a cross link with following options:
+
+- `text`: the display text when the cross reference is has resolved correctly.
```
Comment (hellosnow) [inline Documentation/tutorial/links_and_cross_references.md]: and line 281

---
In reply to: [83810721](https://github.com/dotnet/docfx/pull/796#discussion_r83810721) [](ancestors = 83810721)

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**